src/index.js

```diff
@@ -25,12 +25,16 @@ export default class Email extends Component {
       // Set value and suggestion state by every change
       this.setState({ value: emailAddress, suggestion: suggest}, ()=>this.selectText())
       event.target.value = emailAddress;
-      this.props.onChange(event)
+      if(this.props.onChange){
+        this.props.onChange(event)
+      }
     } else {
       // Update value state plus suggested text
       this.setState({ value: emailAddress + suggest, suggestion: suggest}, ()=>this.selectText())
       event.target.value = emailAddress + suggest
-      this.props.onChange(event)
+      if(this.props.onChange){
+        this.props.onChange(event)
+      }
```

Comment:
Since the `this.props.onChange(event)` is called in all condition of `if`, you can avoid duplication by calling it after `if`.
```js
if (true) {
  // pass
} else {
  // pass
}

this.props.onChange(event)
```

Reply:
Cool!, i'll change it

Reply:
But I still have to add the condition `if (this.props.onchange)` becouse your component doesn't required it as default.

Reply:
Yeah, its okay!